### ISSUE TITLE
Truncated mpo takes forever

### ISSUE THREAD

**Issue by doublex** [python-pillow/Pillow#5864]
Image: http://doppelbauer.name/takesforever.jpg
(do you mind if I delete the image?)

Testcase:
````python
import PIL.Image, PIL.ImageFile
PIL.ImageFile.LOAD_TRUNCATED_IMAGES = True
with PIL.Image.open('takesforever.jpg') as img:
    for frame in PIL.ImageSequence.Iterator(img):
        frame.convert('RGBA')
````

**Comment by radarhere** [comment]
What is your thinking behind deleting the image?
When you say it takes forever, that's not an exaggeration?

**Comment by doublex** [comment]
I have killed the python process after 2 hours (therefore I suspect an infinite loop).
If this issue is wrong, please close it. 

The photo is a user-upload ("avatar") – if there is a problem: could you download this photo to your computer?

Thanks a lot for all your efforts

**Comment by radarhere** [comment]
Hi. I know you probably think we just forgot about this, but not so - this should be resolved in Pillow 9.0, thanks to #5921.

We actually classify images that take forever to load as a security vulnerability - which can lead to reluctance to talk about it before we have put a fix out there.

If you find another such image, would you mind [following the security procedure](https://github.com/python-pillow/Pillow#report-a-vulnerability), rather than creating an issue? Thanks.
